attempt to fix CSL

MikaelSlevinsky opened the pull request:
Skipping the audit appears to prevent rpath from doing anything (https://github.com/JuliaBinaryWrappers/CompilerSupportLibraries_jll.jl/issues/1). But rpath needs some write permissions. There is still
```
Warning: Unable to set SONAME on lib/libgcc_s.1.dylib
```

staticfloat commented on 0_RootFS/CompilerSupportLibraries/build_tarballs.jl:
```diff
@@ -45,6 +45,11 @@ for d in /opt/${target}/${target}/lib*; do
+
+# change permissions so that rpath succeeds
+for l in ${libdir}/*; do
+    chmod a+w "${l}"
```
Let's set this to a default `0755`.

MikaelSlevinsky replied:
Ok I changed this. Still issues the following warnings
```julia
[ Info: Building for x86_64-apple-darwin14-libgfortran3, x86_64-apple-darwin14-libgfortran4, x86_64-apple-darwin14-libgfortran5
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
┌ Warning: Could not probe /Users/Mikael/Codes/build/x86_64-apple-darwin14-libgfortran3/7f3luil5/destdir/lib/libgcc_s.1.dylib for an SONAME!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:9
┌ Warning: ObjectFile.MagicMismatch("Object file is not any of ObjectFile.ELF.ELFHandle, ObjectFile.MachO.MachOHandle, ObjectFile.COFF.COFFHandle! To force one object file format use readmeta(io, T).")
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:10
/docker_entrypoint.sh: line 24: /opt/x86_64-apple-darwin14/bin/install_name_tool: No such file or directory
Warning: Unable to set SONAME on lib/libgcc_s.1.dylib
┌ Warning: Could not probe /Users/Mikael/Codes/build/x86_64-apple-darwin14-libgfortran3/7f3luil5/destdir/lib/libgcc_s.1.dylib for an SONAME!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:9
┌ Warning: ObjectFile.MagicMismatch("Object file is not any of ObjectFile.ELF.ELFHandle, ObjectFile.MachO.MachOHandle, ObjectFile.COFF.COFFHandle! To force one object file format use readmeta(io, T).")
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:10
┌ Warning: Could not probe /Users/Mikael/Codes/build/x86_64-apple-darwin14-libgfortran3/7f3luil5/destdir/lib/libgcc_s_ppc64.1.dylib for an SONAME!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:9
┌ Warning: ObjectFile.MagicMismatch("Object file is not any of ObjectFile.ELF.ELFHandle, ObjectFile.MachO.MachOHandle, ObjectFile.COFF.COFFHandle! To force one object file format use readmeta(io, T).")
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:10
/docker_entrypoint.sh: line 24: /opt/x86_64-apple-darwin14/bin/install_name_tool: No such file or directory
Warning: Unable to set SONAME on lib/libgcc_s.1.dylib
┌ Warning: Could not probe /Users/Mikael/Codes/build/x86_64-apple-darwin14-libgfortran3/7f3luil5/destdir/lib/libgcc_s_ppc64.1.dylib for an SONAME!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:9
┌ Warning: ObjectFile.MagicMismatch("Object file is not any of ObjectFile.ELF.ELFHandle, ObjectFile.MachO.MachOHandle, ObjectFile.COFF.COFFHandle! To force one object file format use readmeta(io, T).")
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:10
┌ Warning: Could not probe /Users/Mikael/Codes/build/x86_64-apple-darwin14-libgfortran3/7f3luil5/destdir/lib/libgcc_s_x86_64.1.dylib for an SONAME!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:9
┌ Warning: ObjectFile.MagicMismatch("Object file is not any of ObjectFile.ELF.ELFHandle, ObjectFile.MachO.MachOHandle, ObjectFile.COFF.COFFHandle! To force one object file format use readmeta(io, T).")
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:10
/docker_entrypoint.sh: line 24: /opt/x86_64-apple-darwin14/bin/install_name_tool: No such file or directory
Warning: Unable to set SONAME on lib/libgcc_s.1.dylib
┌ Warning: Could not probe /Users/Mikael/Codes/build/x86_64-apple-darwin14-libgfortran3/7f3luil5/destdir/lib/libgcc_s_x86_64.1.dylib for an SONAME!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:9
┌ Warning: ObjectFile.MagicMismatch("Object file is not any of ObjectFile.ELF.ELFHandle, ObjectFile.MachO.MachOHandle, ObjectFile.COFF.COFFHandle! To force one object file format use readmeta(io, T).")
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:10
Warning: Unable to find valid license file in "${prefix}/share/licenses/CompilerSupportLibraries"
┌ Warning: Could not probe /Users/Mikael/Codes/build/x86_64-apple-darwin14-libgfortran3/7f3luil5/destdir/lib/libgcc_s.1.dylib for an SONAME!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:9
┌ Warning: ObjectFile.MagicMismatch("Object file is not any of ObjectFile.ELF.ELFHandle, ObjectFile.MachO.MachOHandle, ObjectFile.COFF.COFFHandle! To force one object file format use readmeta(io, T).")
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/CASCi/src/auditor/soname_matching.jl:10

```

MikaelSlevinsky replied:
Note I'm in way over my head in the valley of the shadow death. I don't know how to fix the remaining issue, just thought I could get the ball rolling

MikaelSlevinsky commented:
Hey @staticfloat, I think this is ready for review.

- I removed the lines at the top defining libdir, as I gather this has been added as an environment variable more recently.
- I changed file permissions to default to allow the `install_name_tool` to succeed for all but one lib:
- On MacOS (the only platform I have), I added the special case for `libgcc_s.1.dylib`.

To test the new product, I copied over all the new libs into the `lib` folder of the v0.2.0+0 `CompilerSupportLibraries_jll` artifact... and `using` it didn't throw the InitError (https://github.com/JuliaBinaryWrappers/CompilerSupportLibraries_jll.jl/issues/1)!

The warnings in the build persist, and it's not particularly elegant, but it works.

staticfloat commented:
Great stuff @MikaelSlevinsky.  I'm going to add a license installation line, merge and manually update these.
